### PR TITLE
fix(gatsby-source-contentful): Correct supported image formats

### PR DESCRIPTION
**Pull request by ascorbic**
Contentful doesn't support AVIF. This fixes the resolver, and adds a check in the url builder

### CODE REVIEW

File: packages/gatsby-source-contentful/src/extend-node-type.js [hunk 1]
```diff
@@ -33,6 +34,8 @@ const {
 // cache is more likely to go stale than the images (which never go stale)
 // Note that the same image might be requested multiple times in the same run
 
+const validImageFormats = new Set([`jpg`, `png`, `webp`])
```
Comment (wardpeet) [inline packages/gatsby-source-contentful/src/extend-node-type.js:37]: Not sure why a set is needed here.
```suggestion
const validImageFormats = [`jpg`, `png`, `webp`]
```

File: packages/gatsby-source-contentful/src/extend-node-type.js [hunk 2]
```diff
@@ -200,6 +203,13 @@ const generateImageSource = (
     height = CONTENTFUL_IMAGE_MAX_SIZE
   }
 
+  if (!validImageFormats.has(toFormat)) {
```
Comment (wardpeet) [inline packages/gatsby-source-contentful/src/extend-node-type.js:206]:
```suggestion
  if (!validImageFormats.includes(toFormat)) {
```

**Review by wardpeet — APPROVED** [review]
I've added a small nit.

File: packages/gatsby-source-contentful/src/extend-node-type.js (hunk 1 above)
Comment (ascorbic) [reply packages/gatsby-source-contentful/src/extend-node-type.js:37]: I was under the impression that set.has was significantly faster than array.includes

Comment (wardpeet) [reply packages/gatsby-source-contentful/src/extend-node-type.js:37]: It probably is but it doesn't matter on a 3 items list. For me when I read Sets I expect things to get added to it and making sure things are unique and that makes it more complex in my head.

Sets are fine, that's why I approved it but wanted to get the reason on the why :)

Comment (ascorbic) [reply packages/gatsby-source-contentful/src/extend-node-type.js:37]: It's just a pattern I always use for this sort of check, particularly when dealing with potentially large numbers of nodes, with a constant Set.

**Comment by ascorbic** [comment]
Published in `gatsby-source-contentful@4.6.2`

**Pull request merged**